### ISSUE TITLE
Proposal: Module renaming for v7

### ISSUE THREAD

**Issue by rowanwins** [Turfjs/turf#2163]
If we're working towards a release of v7 then now would be a good time to rename any modules as a breaking change.

- [ ] `lineIntersect` to become `intersectionPoints`. Rationale being is that this module supports both polygons and lines, and I think the proposed name is a bit clearer as to what it does and returns.

Any other thoughts for candidates @mfedderly @JamesLMilner @twelch ?

**Comment by JamesLMilner** [comment]
I agree that lineIntersect is a bit confusing considering it works on LineStrings and Polygons. I would be happy to rename it to intersectionPoints if that helps provide clarity to users

**Comment by mfedderly** [comment]
I don't know of any other renames offhand, but I made a PR for lineIntersect https://github.com/Turfjs/turf/pull/2187
